crypto/pkcs11key/key.go

```diff
@@ -30,6 +30,24 @@ var hashPrefixes = map[crypto.Hash][]byte{
 	crypto.RIPEMD160: {0x30, 0x20, 0x30, 0x08, 0x06, 0x06, 0x28, 0xcf, 0x06, 0x03, 0x00, 0x31, 0x04, 0x14},
 }
 
+// Ctx defines the subset of pkcs11.Ctx's methods that we use, so we can inject
+// a different Ctx for testing.
+type Ctx interface {
```

Comment:
I thought of another argument against telling clients to load the library in `init()` and pass a pointer. If we load the library in `init()`, the best we can do is panic. However, if we do it later during the application, we can return an error. That allows the app to log however it likes, and choose between a clean shutdown or continuing to run with some features disabled (if PKCS#11 support is not a core part of the application).

Also, in general the module path will be a runtime config option, so it's nice to give the app time to load config files before initializing the module.

If we think that in general clients will want to initialize the module outside of `init()`, they start having the same problem of ensuring that the module only loads once, which means reimplementing locking code similar to what we have here.